### PR TITLE
[rpc] Fix pruneheight help description in getblockchaininfo

### PR DESCRIPTION
**Pull request by esotericnonsense**
This is a bit nitpicky, but the field isn't included in the JSON at all unless fPruneMode is set.
Alternatively we could return 0 (and remove 'pruned', which would then be pointless, but someone might be relying on that).

### CODE REVIEW

**Comment by practicalswift** [comment]
utACK 9da936232743a2235dade998be7f01ce0cf1b709. Clarity is good.

**Comment by jnewbery** [comment]
utACK 9da936232743a2235dade998be7f01ce0cf1b709

Also consider in this PR moving the `if (fPruneMode)` code block up in the function so that it's immediately below the `pruned` line. That seems more logical and the return object would be ordered the same as the help text.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1136,7 +1136,7 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
             "  \"verificationprogress\": xxxx, (numeric) estimate of verification progress [0..1]\n"
             "  \"chainwork\": \"xxxx\"     (string) total amount of work in active chain, in hexadecimal\n"
             "  \"pruned\": xx,             (boolean) if the blocks are subject to pruning\n"
-            "  \"pruneheight\": xxxxxx,    (numeric) lowest-height complete block stored\n"
+            "  \"pruneheight\": xxxxxx,    (numeric) lowest-height complete block stored (if pruning is enabled)\n"
```
Comment (promag) [inline src/rpc/blockchain.cpp]: Update to `(only present if pruning is enabled)`. See [this example](https://github.com/bitcoin/bitcoin/blob/791a0e6ddade27d1b69f4861a6640de60b9553cf/src/wallet/rpcwallet.cpp#L1845) for reference.

**Review by promag — COMMENTED** [review]
Nit, I would reword commit and PR to something along `[rpc] Fix pruneheight help description in getblockchaininfo`.

Otherwise utACK 9da9362.

**Comment by esotericnonsense** [comment]
Done and done, tested latest commit, appears in correct order.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -1180,6 +1180,14 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
     obj.push_back(Pair("verificationprogress",  GuessVerificationProgress(Params().TxData(), chainActive.Tip())));
     obj.push_back(Pair("chainwork",             chainActive.Tip()->nChainWork.GetHex()));
     obj.push_back(Pair("pruned",                fPruneMode));
+    if (fPruneMode)
```
Comment (promag) [inline src/rpc/blockchain.cpp]: Since this is touched, update as per developer notes? `{` in this line.

File: src/rpc/blockchain.cpp [hunk 3]
```diff
@@ -1180,6 +1180,14 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
...
+    {
+        CBlockIndex *block = chainActive.Tip();
+        while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
```
Comment (promag) [inline src/rpc/blockchain.cpp]: Add `{ }`.

Comment (promag) [reply src/rpc/blockchain.cpp]: Remove first condition `block` and add `assert(block);` above.

**Comment by esotericnonsense** [comment]
Suggest merging this with #11367 to avoid dependencies?

**Comment by esotericnonsense** [comment]
Changes are now included in #11367.

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (esotericnonsense) [reply src/rpc/blockchain.cpp]: Done in #11367
